proto/scene-format.proto

```diff
@@ -0,0 +1,62 @@
+syntax = "proto3";
+
+package turbocow;
+
+message Scene {
+	enum MaterialType {
+		SOLID = 0;
+		GLASS = 1;
+		// other matherials
+	}
+	message Vector3 {
+		float x = 2;
+		float y = 3;
+		float z = 4;
+	}
+	message Transform {
+		int32 id = 5;
+		Vector3 translation = 6;
+		Vector3 rotation = 7;
+		Vector3 scale = 8;
+		int32 parent_id = 9;
+	}
+	message Cube {
+		int32 id = 34;
+		Vector3 upper_bounds = 10;
+		Vector3 lower_bounds = 11;
+		Transform transform = 11;
+	}
+	message Sphere {
+		int32 id = 35;
+		Vector3 positon = 12;
+		float radius = 13;
+	}
+	message MeshedObject {
+		int32 id = 14;
+		string reference = 15;
+		MaterialType matherial_type = 16;
+		Transform transform = 17;
+	}
+	enum LightType {
+		POINT = 18;
+		SPHERE = 19;
+		DIRECTIONAL = 20;
+	}
+	message Light {
+		int32 id = 21;
+		LightType light_type = 22;
+		Transform transform = 23;
+		Vector3 color = 24;
+		double intensity = 25;
+	}
+	message Camera {
+		int32 id = 26;
+		Transform transform = 27;
+	}
+	repeated Object objects = 28;
+	optional repeated Cube cubes = 29;
+	optional repeated Sphere spheres = 30;
```

Comment:
What about grouping objects into one type, `SceneObject`? Like this:
```
message SceneObject {
    int32 id = 1;
    Transform transform = 2;
    oneof obj {
        Sphere sphere = 3;
        Cube cube = 4;
        MeshedObject meshedObject = 5;
    }
}
```

Then we can have a single repeated field of `SceneObject`s instead of a separate field for each primitive type.

Reply:
It is a good idea to group them, but I think `mesh` is a better name than `obj`. And also `material` should be specified

Reply:
Thanks, good idea